app/javascript/packs/billboard.js

```diff
@@ -14,20 +14,25 @@ async function generateDisplayAd(element) {
   const { asyncUrl } = element.dataset;
 
   if (asyncUrl) {
-    const response = await window.fetch(`${asyncUrl}`);
-    const htmlContent = await response.text();
+    try {
+      const response = await window.fetch(`${asyncUrl}`);
+      const htmlContent = await response.text();
 
-    const generatedElement = document.createElement('div');
-    generatedElement.innerHTML = htmlContent;
+      const generatedElement = document.createElement('div');
+      generatedElement.innerHTML = htmlContent;
 
-    element.innerHTML = '';
-    element.appendChild(generatedElement);
-    setupDisplayAdDropdown();
-    // This is called here because the ad is loaded asynchronously.
-    // The original code is still in the asset pipeline, so is not importable.
-    // This could be refactored to be importable as we continue that migration.
-    // eslint-disable-next-line no-undef
-    observeDisplayAds();
+      element.innerHTML = '';
+      element.appendChild(generatedElement);
+      setupDisplayAdDropdown();
+      // This is called here because the ad is loaded asynchronously.
+      // The original code is still in the asset pipeline, so is not importable.
+      // This could be refactored to be importable as we continue that migration.
+      // eslint-disable-next-line no-undef
+    } catch (error) {
+      if (!/NetworkError/i.test(error.message)) {
+        Honeybadger.notify(error);
+      }
```

Comment:
So I can't write a unit test for this pack file because the function doesn't need to be exported, and I don't think there's cypress spec testing the display of billboard yet, I'm going to bypass this test requirement.